dlt/resource_dlt_pipeline.go

```diff
@@ -0,0 +1,357 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	// "github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// DefaultTimeout is the default amount of time that Terraform will wait when creating, updating and deleting pipelines.
+const DefaultTimeout = 20 * time.Minute
+
+type createPipelineRequestStruct struct {
+	pipelines.CreatePipeline
+}
+
+type updatePipelineRequestStruct struct {
+	pipelines.EditPipeline
+}
+
+// Name not keeping twice
+type Pipeline struct {
+	pipelines.PipelineSpec
+	PipelineId          string                              `json:"pipeline_id,omitempty"`
```

Comment:
Some of these things are computed, correct? I know it's not very well documented, but we can try to mark them as such.

Reply:
Also, in particular fot the `pipeline_id`, this is an alias for `id`, correct?

Reply:
So basically the get response returns a `pipeline_id` as well as a `spec` object which contains `id`. But they are essentially the same thing

Reply:
Then I think it's better to use the alias option